src/openvic-simulation/modifier/StaticModifierCache.cpp

```diff
@@ -1,115 +1,172 @@
+					return true; //an event modifier overrides a static modifier with the same identifier.
+				}
 
-		if (mutable_modifier != nullptr) {
-			if (multiplier != fixed_point_t::_1()) {
-				*mutable_modifier *= multiplier;
+				ModifierValue modifier_value {};
+				bool has_parsed_modifier = expect_dictionary(modifier_manager.expect_base_country_modifier(modifier_value))(value);
+				has_parsed_modifier &= modifier_manager.add_event_modifier(identifier, std::move(modifier_value), icon, STATIC);
+				return has_parsed_modifier;
 			}
+		);
+	
+	};
 
-			modifier = mutable_modifier;
-		} else {
-			Logger::error("Failed to set static modifier: ", name);
-			ret = false;
-		}
+	const auto set_static_province_modifier = [&key_map, &modifier_manager, &ret](
+		Modifier& modifier
+	) -> void {
+		ret &= add_key_map_entry(
+			key_map, modifier.get_identifier(), ONE_EXACTLY,
+			expect_dictionary(modifier_manager.expect_base_province_modifier(modifier))
+		);
 	};
 
 	// Country modifiers
-	set_static_modifier(very_easy_player, "very_easy_player");
-	set_static_modifier(easy_player, "easy_player");
-	set_static_modifier(hard_player, "hard_player");
-	set_static_modifier(very_hard_player, "very_hard_player");
-	set_static_modifier(very_easy_ai, "very_easy_ai");
-	set_static_modifier(easy_ai, "easy_ai");
-	set_static_modifier(hard_ai, "hard_ai");
-	set_static_modifier(very_hard_ai, "very_hard_ai");
-
-	set_static_modifier(base_modifier, "base_values");
-	set_static_modifier(war, "war");
-	set_static_modifier(peace, "peace");
-	set_static_modifier(disarming, "disarming");
-	set_static_modifier(war_exhaustion, "war_exhaustion");
-	set_static_modifier(infamy, "badboy");
-	set_static_modifier(debt_default_to, "debt_default_to");
-	set_static_modifier(bad_debter, "bad_debter");
-	set_static_modifier(great_power, "great_power");
-	set_static_modifier(secondary_power, "second_power");
-	set_static_modifier(civilised, "civ_nation");
-	set_static_modifier(uncivilised, "unciv_nation");
-	set_static_modifier(literacy, "average_literacy");
-	set_static_modifier(plurality, "plurality");
-	set_static_modifier(generalised_debt_default, "generalised_debt_default");
-	set_static_modifier(total_occupation, "total_occupation", 100);
-	set_static_modifier(total_blockaded, "total_blockaded");
-	set_static_modifier(in_bankruptcy, "in_bankrupcy");
+	set_static_country_modifier(very_easy_player);
+	set_static_country_modifier(easy_player);
+	set_static_country_modifier(hard_player);
+	set_static_country_modifier(very_hard_player);
+	set_static_country_modifier(very_easy_ai);
+	set_static_country_modifier(easy_ai);
+	set_static_country_modifier(hard_ai);
+	set_static_country_modifier(very_hard_ai);
+
+	ret &= add_key_map_entry(
+		key_map, base_modifier.get_identifier(), ONE_EXACTLY,
+		expect_dictionary_keys_and_default(
+			modifier_manager.expect_base_country_modifier(base_modifier),
+			"supply_limit", ZERO_OR_ONE, [this, &modifier_manager](auto const& value) -> bool {
```

Comment:
```suggestion
			"supply_limit", ZERO_OR_ONE, [this, &modifier_manager](ast::NodeCPtr value) -> bool {
```
Change to explicit type rather than auto const reference